src/store/selection/actions.ts

```diff
@@ -27,10 +28,11 @@ export const setSelectionAction = createAction(
             };
         }
 
-        const { location, range } = arg;
+        const { containerRect, location, range } = arg;
 
         return {
             payload: {
+                containerRect: getShape(containerRect),
```

Comment:
Can we store the rects in a normalized state upon selection instead of storing additional state and doing so later? That is, can we call the combine/dedupe/etc. helpers in this action for `rects`, instead?